functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,637 @@
+			Expect(err).NotTo(HaveOccurred())
+
+			value, ok, err := unstructured.NestedString(object.Object, "metadata", amInfo.metadata.kind, "mutated")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ok).To(BeTrue())
+			Expect(value).To(Equal("true"))
+
+			err = client.Resource(object.resourceName).Namespace(object.GetNamespace()).Delete(context.Background(), object.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		},
+		createTableEntries("added", []metadataInfo{
+			{
+				kind:  "labels",
+				key:   "mutated",
+				value: "true",
+			},
+			{
+				kind:  "annotations",
+				key:   "mutated",
+				value: "true",
+			},
+		}...)...,
+	)
+
+	DescribeTable("should not mutate existing metadata info(labels/annotations)",
+		func(amInfo assignMetadataInfo, object TestObject) {
+			amDefinition := assignMetadataDefinition(&amInfo)
+
+			amDefinition, err := client.Resource("assignmetadata").Create(context.Background(), amDefinition, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			defer func() {
+				err := client.Resource("assignmetadata").Delete(context.Background(), amDefinition.GetName(), metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}()
+
+			unstructured.SetNestedField(object.Object, "false", "metadata", amInfo.metadata.kind, "changed")
+
+			object.Unstructured, err = client.Resource(object.resourceName).Namespace(object.GetNamespace()).Create(context.Background(), object.Unstructured, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			value, ok, err := unstructured.NestedString(object.Object, "metadata", amInfo.metadata.kind, "changed")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ok).To(BeTrue())
+			Expect(value).To(Equal("false"))
+
+			err = client.Resource(object.resourceName).Namespace(object.GetNamespace()).Delete(context.Background(), object.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		},
+		createTableEntries("exist", []metadataInfo{
+			{
+				kind:  "labels",
+				key:   "changed",
+				value: "true",
+			},
+			{
+				kind:  "annotations",
+				key:   "changed",
+				value: "true",
+			},
+		}...)...,
+	)
+
+	It("should apply mutations by order", func() {
+		amDefB := &assignMetadataInfo{
+			name:  "mutation-b",
+			kinds: []string{"Pod"},
+			scope: "Namespaced",
+			metadata: metadataInfo{
+				kind:  "labels",
+				key:   "mutated-by",
+				value: "b",
+			},
+		}
+
+		By("Creating assignMetadata mutation-b")
+		amB := assignMetadataDefinition(amDefB)
+		amB, err := client.Resource("assignmetadata").Create(context.Background(), amB, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), amB.GetName(), metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		By("Creating test-pod-b")
+		testPodB := GetTestPod(TestingNamespace)
```

Comment:
I think we already have a create pod function no?

Reply:
i needed a customized pod for the tests, also i needed it to be unstructured so that tests can be generic.